sql/types/conversion.go

```diff
@@ -552,3 +552,76 @@ func TypesEqual(a, b sql.Type) bool {
 		return a == b
 	}
 }
+
+func BiggerType(a, b sql.Type) sql.Type {
```

Comment:
does `sql.Type.Promote` simplify any of this/more future proof? We have "to biggest" type coercion in other places also, like `mergeSetOpSchemas`, do we need two paths for this?